src/Target.cpp

```diff
@@ -555,10 +599,11 @@ void bad_target_string(const std::string &target) {
         }
     }
     user_error << "Did not understand Halide target " << target << "\n"
-               << "Expected format is arch-bits-os-feature1-feature2-...\n"
+               << "Expected format is arch-bits-os-processor-feature1-feature2-...\n"
```

Comment:
This implies that "processor" is a mandatory field. I'd just leave it out here. To a user it acts like a feature flag, except that you can only set one of them.

Reply:
I think it is pretty important for this to not invalidate existing target strings. We definitely need to accept strings with no processor specified on input. (I'm guessing that is already the case but have not read the entire PR.) Open question whether output should omit the processor if it is generic. Strictly speaking, always inserting it will likely break something so we would need to phase that in if we are serious about backward compatibility.

Reply:
> I think it is pretty important for this to not invalidate existing target strings. We definitely need to accept strings with no processor specified on input. (I'm guessing that is already the case but have not read the entire PR.)

Right, (at least currently?) it's not required, and defaults to generic,
which results in feature-driven tune CPU selection.

> Open question whether output should omit the processor if it is generic. Strictly speaking, always inserting it will likely break something so we would need to phase that in if we are serious about backward compatibility.

Well, should we omit or not? :)

Reply:
I vote we omit it printing it when it's generic, and we also omit it from this error string, because as it stands it implies processor is a required field.

Reply:
Right, since `host` feature is expanded, i guess not printing `tune_generic` is safe to do.

Though, i think we should still list the known/supported CPU's, no?
How about just explicitly saying here that CPU is optional?